includes/Checker/Checks/Trademarks_Check.php

```diff
@@ -0,0 +1,482 @@
+		'internet-explorer-',
+		'ios-',
+		'jetpack-',
+		'macintosh-',
+		'macos-',
+		'mailchimp-',
+		'microsoft-',
+		'ninja-forms-',
+		'oculus',
+		'onlyfans-',
+		'only-fans-',
+		'opera-',
+		'paddle-',
+		'paypal-',
+		'pinterest-',
+		'plugin',
+		'skype-',
+		'stripe-',
+		'tiktok-',
+		'tik-tok-',
+		'trustpilot',
+		'twitch-',
+		'twitter-',
+		'tweet',
+		'ups-',
+		'usps-',
+		'vvhatsapp',
+		'vvcommerce',
+		'vva-',
+		'vvoo',
+		'wa-',
+		'webpush-vn',
+		'wh4tsapps',
+		'whatsapp',
+		'whats-app',
+		'watson',
+		'windows-',
+		'wocommerce',
+		'woocom-',
+		'woocommerce',  // technically ending with '-for-woocommerce' is allowed.
+		'woocomerce',
+		'woo-commerce',
+		'woo-',
+		'wo-',
+		'wordpress',
+		'wordpess',
+		'wpress',
+		'wp-',
+		'wp-mail-smtp-',
+		'yandex-',
+		'yahoo-',
+		'yoast',
+		'youtube-',
+		'you-tube-',
+	);
+
+	/**
+	 * Lists of trademarks that are allowed as 'for-whatever' ONLY.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @var string[]
+	 */
+	const FOR_USE_EXCEPTIONS = array(
+		'woocommerce',
+	);
+
+	/**
+	 * Lists of commonly used 'combo' names (to prevent things like 'woopress').
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @var string[]
+	 */
+	const PORTMANTEAUS = array(
+		'woo',
+	);
+
+	const CHECK_README = 1;
+	const CHECK_NAME   = 2;
+	const CHECK_SLUG   = 4;
+	const TYPE_ALL     = 7; // Same as all of the above with bitwise OR.
+
+	/**
+	 * Bitwise flags to control check behavior.
+	 *
+	 * @since n.e.x.t
+	 * @var int
+	 */
+	protected $flags = 0;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $flags Bitwise flags to control check behavior.
+	 */
+	public function __construct( $flags = self::TYPE_ALL ) {
+		$this->flags = $flags;
+	}
+
+	/**
+	 * Gets the categories for the check.
+	 *
+	 * Every check must have at least one category.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array The categories for the check.
+	 */
+	public function get_categories() {
+		return array( Check_Categories::CATEGORY_PLUGIN_REPO );
+	}
+
+	/**
+	 * Check for trademarks.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 * @param array        $files  Array of plugin files.
+	 */
+	protected function check_files( Check_Result $result, array $files ) {
+
+		// Check the trademarks in readme file plugin name.
+		if ( $this->flags & self::CHECK_README ) {
+			$this->check_for_readme( $result, $files );
+		}
+
+		// Check the trademarks in plugin name.
+		if ( $this->flags & self::CHECK_NAME ) {
+			$this->check_for_name( $result );
+		}
+
+		// Check the trademarks in plugin slug.
+		if ( $this->flags & self::CHECK_SLUG ) {
+			$this->check_for_slug( $result );
+		}
+	}
+
+	/**
+	 * Checks the trademarks in readme file plugin name.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 * @param array        $files  Array of plugin files.
+	 */
+	private function check_for_readme( Check_Result $result, array $files ) {
```

Comment:
The following code is basically duplicated from the readme check.

Should we just move the trademark check to the `Readme_Check`? Or must it be separate?

If separate, can we perhaps find a way to reuse the logic to retrieve the readme file?

Reply:
Thank you, @swissspidy, for bringing this up. Would it be possible to handle this refactoring in a follow-up PR to avoid making changes to other files within this current PR?

Reply:
I think first we would need to decide on the approach: keep two checks (but with shared code), or merge the checks into one. If the latter, it doesn't really make sense to do it in a follow-up PR. Let's hear others :)